hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/utils/HiveAvroSerializer.java

```diff
@@ -75,17 +76,16 @@ public class HiveAvroSerializer {
   private final List<String> columnNames;
   private final List<TypeInfo> columnTypes;
   private final ArrayWritableObjectInspector objectInspector;
-  private final Schema recordSchema;
+  private final HoodieSchema recordSchema;
 
   private static final Logger LOG = LoggerFactory.getLogger(HiveAvroSerializer.class);
 
-  public HiveAvroSerializer(Schema schema) {
-    schema = AvroSchemaUtils.getNonNullTypeFromUnion(schema);
```

Comment:
Similarly, is there still usage of `AvroSchemaUtils.getNonNullTypeFromUnion` before it can be removed?